### PR TITLE
Update to signalr-1.0.0-preview1-final

### PR DESCRIPTION
**Pull request by DarkCow**
Hello,

I wanted to play & learn with SignalR.Orleans and upgrade Signalr to 1.0.0-preview1-final. I don't expect this to be pulled without first discussing these points with people more familiar with the project.

- [x] I am unsure when the function `SendGroupExceptAsync` was added to SignalR core. And unsure how best to implement it here. I added it as a function to `GroupGrain` instead of `ConnectionGrain`, not sure which is best.
- [x] I used Task.WhenAll to implement the new plural methods ` SendGroups`, `SendUsers` etc... I am unsure if we want to take the more involved approach using a [dispatcher](https://github.com/OrleansContrib/DesignPatterns/blob/master/Dispatcher.md) edit: or even if a dispatcher would improve performance at all.
- [ ] I have made it pass unit test, but unsure what local testing needs to be performed first. So please test for regressions on your end.

### CODE REVIEW

**Comment by galvesribeiro** [comment]
Hello @DarkCow! Thanks for the PR.

We didn't updated yet to preview1 as it depends on .Net Core 2.1 which wasn't released yet.

**Comment by DarkCow** [comment]
When I have time, I can add the unit tests for `SendGroupExceptAsync`, `SendGroups`, `SendUsers`, etc...

**Comment by DarkCow** [comment]
I may be wrong here, but I am running Signalr 1.0.0-preview1-final on a .netcore 2.0 project.

```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    ...
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.2" />
    <PackageReference Include="Microsoft.AspNetCore.SignalR" Version="1.0.0-preview1-final" />
    <PackageReference Include="SignalR.Orleans" Version="1.0.0-preview-3" />
    ...
  </ItemGroup>
</Project>
```

I should admit, I am still working on my code...

**Comment by galvesribeiro** [comment]
Funny... Someone told me from the .Net Team that it was `netstandard2.1`. I just checked the dependencies on nuget.org and you are right.

@claylaut can you have a look at this when you have spare time?

**Comment by claylaut** [comment]
@DarkCow / @galvesribeiro,

Even me I thought it was switched to .NET Core 2.1 Preview (for sure there repo is on 2.1 preview since last time I was trying to build it and it failed because I don't have .net core 2.1)

We shall proceed then and update it :)

regarding your questions:
1. SendGroupExceptAsync Implementation:
Yes,  it should be in `GroupGrain` not in `ConnectionGrain` since this functionality only applies for Group and not for the Users.

2. Using Task.WhenAll, yes thats the correct approach. 

Regarding the Dispatcher pattern, i m interested to see the implementation you have in mind. if you have some free time you can open a seperate pr with this change. @galvesribeiro what do you think?

**Comment by galvesribeiro** [comment]
I agree. We should leave the dispatch pattern to a separated PR. So, please continue implementing this one and once it is ready for review let us know. Thank you!

File: src/SignalR.Orleans/Core/GrainExtensions.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static class GrainSignalRExtensions
     {
         public static async Task SendSignalRMessage(this IConnectionGrain grain, string methodName, params object[] message)
         {
-            var invocationMessage = new InvocationMessage(Guid.NewGuid().ToString(), nonBlocking: true, target: methodName, arguments: message);
+            var invocationMessage = new InvocationMessage(target: methodName, argumentBindingException: null, arguments: message);
```
Comment (claylaut) [inline src/SignalR.Orleans/Core/GrainExtensions.cs]: why did you remove the invocationId?

File: src/SignalR.Orleans/OrleansHubLifetimeManager.cs [hunk 2]
```diff
@@ -81,32 +81,38 @@ public override Task AddGroupAsync(string connectionId, string groupName)
...
             var connection = this._connections[connectionId];
-            if (connection != null) return InvokeLocal(connection, message);
+            if (connection != null) return SendLocal(connection, message);
 
-            return InvokeExternal(connectionId, message);
+            return SendExternal(connectionId, message);
         }
 
-        public override Task InvokeGroupAsync(string groupName, string methodName, object[] args)
+        public override Task SendConnectionsAsync(IReadOnlyList<string> connectionIds, string methodName, object[ ] args)
```
Comment (claylaut) [inline src/SignalR.Orleans/OrleansHubLifetimeManager.cs]: Can you do autoformat or fix styling for `object[ ]` to `object[]` please :)

File: src/SignalR.Orleans/Core/GrainExtensions.cs (hunk 1 above)
Comment (DarkCow) [reply src/SignalR.Orleans/Core/GrainExtensions.cs]: Looking through SignalR's PRs I found talk of removing invocation ID... https://github.com/aspnet/SignalR/pull/1218/files

specifically 

https://github.com/aspnet/SignalR/commit/00a6dc983ac1df1ddc51c968c060bff8f45e5b2e#diff-78e1f2ad8780817477e28d3ec3b11ff6

I might have a misunderstanding though.... It appears that they've change it yet again pending next release.
https://github.com/aspnet/SignalR/pull/1843/files

**Comment by galvesribeiro** [comment]
So, where we are on this PR? Is it ok to review/merge?

Ping @claylaut

File: src/SignalR.Orleans/Core/GrainExtensions.cs (hunk 1 above)
Comment (claylaut) [reply src/SignalR.Orleans/Core/GrainExtensions.cs]: Thanks for your answers, in one of the PR's you've linked there is below (sharing it for general knowledge):
Non-Blocking Invocations
+Invocations can be sent without an `Invocation ID` value. This indicates that the invocation is "non-blocking", and thus the caller does not expect a response. When a Callee receives an invocation without an `Invocation ID` value, it **must not** send any response to that invocation.

**Comment by claylaut** [comment]
@galvesribeiro,

It looks good to me.
